### ISSUE TITLE
Remove polyfill packages

### ISSUE THREAD

**Issue by thompson-tomo** [App-vNext/Polly#2057]
**Is your feature request related to a problem? Please describe.**
I want to minimise dependencies in my project by utilising framework dependencies wherever possible

**Describe the solution you'd like**
In Polly.Extensions System.Diagnostics.DiagnosticSource can be provided via the framework ie net 6/8 as such conditions should be placed on it so that it is only included for the other frameworks. Other dependencies can be optimised.

**Describe alternatives you've considered**
Accept the additional dependency

**Additional context**
n/a

**Comment by martincostello** [comment]
I'm lead to believe that removing dependencies is a breaking change (in fact, I hit this today in #2003 due to changes to Microsoft.Extensions.TimeProvider.Testing dropping seemingly redundant dependencies), so if that's the case this isn't something we'd do until there's a reason for a Polly v9 if we did.

**Comment by thompson-tomo** [comment]
In the case here I am referring to polyfill package's ie ones which are natively provided by the framework without it needing to be explicitly added hence I don't see it as a breaking change.

**Comment by martincostello** [comment]
Yes I understand that, and this comment is what lead me to my comment: https://github.com/dotnet/extensions/pull/5058#pullrequestreview-1947513547

**Comment by thompson-tomo** [comment]
I am still talking about polyfill package's and as such if a client is using System.Diagnostic.DiagnosticSource in their application/library it will still build as the framework will provide the library and likely even provide a newer version.

**Comment by martintmk** [comment]
The `System.Diagnostics.DiagnosticSource` is only included on `netstandard2.0` `net472` `net462` and we cannot really use the one that is provided by .NET Framework. This is because we are using APIs that are only available on newer version of the library.

If I add `System.Diagnostics.DiagnosticSource` as per your suggestion:

``` xml
  <ItemGroup>
    <PackageReference Include="Microsoft.Extensions.Options" />
    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" />
    <Reference Include="System.Net.Http" Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'" />
    <Reference Include="System.Diagnostics.DiagnosticSource" Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'" />
  </ItemGroup>
```

The `Polly.Extensions` cannot be built anymore:

<img width="1012" alt="image" src="https://github.com/App-vNext/Polly/assets/103487740/e05c2688-ba38-4b23-b25c-44896e49e669">

**Comment by thompson-tomo** [comment]
Issue has been addressed already as part of the 8.4.0 release which has occured, hence closing issue
